src/raygun.rum/index.js

```diff
@@ -958,6 +980,76 @@ var raygunRumFactory = function(window, $, Raygun) {
       makePostCorsRequest(url, stringifiedPayload, successCallback, errorCallback);
     }
 
+    function updateUserAgentData(data) {
+      if (!data.eventData) { return; }
+
+      for (var i = 0; i < data.eventData.length; i++) {
+        if (!!data.eventData[i].data && Array.isArray(data.eventData[i].data)) {
+
+          if (!!data.eventData[i].device) {
+            data.eventData[i].device = getHighFidelityUAString(data.eventData[i].device);
+          }
+
+          for (var k = 0; k < data.eventData[i].data.length; k++) {
+            var ek = data.eventData[i].data[k];
+
+            if (!!ek.userAgent) {
+              ek.userAgent = getHighFidelityUAString(ek.userAgent);
+            }
+
+            if (!!window.raygunUserAgentData.platformVersion) {
+              var platformVersion = window.raygunUserAgentData.platformVersion.split(".");
+              ek.device = {
+                Family: window.raygunUserAgentData.platform,
+                Major: platformVersion[0] || '',
+                Minor: platformVersion[1] || '',
+                Patch: platformVersion[2] || '',
+                PatchMinor: platformVersion[3] || ''
+              };
+            }
+          }
+        }
+      }
+    }
+
+    function getHighFidelityUAString(userAgentString) {
+
+      if (!window.raygunUserAgentData) {
+        return userAgentString;
+      }
+      if (window.raygunUserAgentData.platform === "Windows" && window.raygunUserAgentData.platformVersion === "14.0.0") {
+        userAgentString = userAgentString.replace('Windows NT 10.0', 'Windows NT 11.0');
+      }
+      var fullVersionList = window.raygunUserAgentData.fullVersionList;
+
+      if (!fullVersionList) {
+        return userAgentString;
+      }
+
+      var regexChrome = /Chrome\/(\d+)\.(\d+)\.(\d+)\.(\d+)/i;
+      var regexEdge = /Edg\/(\d+)\.(\d+)\.(\d+)\.(\d+)/i;
+      // var regexOpera = /OPR\/(\d+)\.(\d+)\.(\d+)\.(\d+)/i; // Not used below, yet???
+
+      for (var n = 0; n < fullVersionList.length; n++) {
+
+        var version = fullVersionList[n].version;
+        var brand = fullVersionList[n].brand;
+
+        if (brand === "Chromium") {
+          userAgentString = userAgentString.replace(regexChrome, 'Chrome\/' + version);
+        }
+        if (brand === "Microsoft Edge") {
+          userAgentString = userAgentString.replace(regexEdge, 'Edg\/' + version);
+        }
+        if (brand === "Opera") {
+          //opera version 88 behaves differently, correctly populating the UserAgent string but not the high entropy UserAgentData. This may change in the future?!               
```

Comment:
Can we resolve these comments in the mean time? Either by leaving them in with the minimal data or just removing them?

Reply:
I would like to leave them in, I expect the behavior to, maybe, change soon.